app.rb

```diff
@@ -0,0 +1,102 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'Book list'
+    @books.each { |book| puts "title : #{book.title} by author : #{book.author}" }
+  end
+
+  def list_persons
+    @persons.each do |person|
+      if person.instance_of?(Student)
+        puts "[Student]Age: #{person.age}, Name: #{person.name}, ID: #{person.id} "
+      else
+        puts "[Teacher]Age: #{person.age}, Name: #{person.name}, ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to add a 1)student or 2)teacher? [Insert 1 or 2]: '
+    is_student = gets.chomp.strip.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    case is_student
+
+    when 1
+      puts 'Has Parent Permission [Y/N]'
+      permission = gets.chomp.downcase == 'y'
+      create_student(age, name, permission)
+    when 2
+      create_teacher(age, name)
+    end
+    puts 'Person added successfully'
+    $stdout.flush
+  end
```

Comment:
- [ ] When I'm trying to create a person(student or teacher) or list all persons, I'm getting errors on the console(sharing one error below re to create person). Please make sure to fix these issues and users should be able to create and list all entities without any errors. (Because of the issue I couldn't test rentals, Please also make sure they work as expected).

![image](https://github.com/supreeti/OPPS-Library-Management-System/assets/83803414/22cbf16b-914f-4db2-8ed1-dda625dc7a1f)